src/ms3/bs4_parser.py

```diff
@@ -2087,8 +2088,28 @@ def get_instrument_name(self, staff_name):
         else:
             return fields_data[staff_name]['trackName']
 
-    def set_instrument(self, staff, trackname):
-        pass
+    def set_instrument(self, staff_name, trackname):
+        if staff_name not in self.parsed_parts.staff2part.keys():
+            raise KeyError(f"Don't recognize key '{staff_name}'")
+        existing_value = self.get_instrument_name(staff_name)
+        new_value = str(trackname)
+        if existing_value is not None and existing_value == new_value:
+            self.logger.debug(f"The {staff_name} was already '{existing_value}' and doesn't need changing.")
+            return
+        new_values = self.INSTRUMENT_DEFAULTS[trackname]
+        for field_to_change in self.instrumentation_fields:
+            value = new_values[field_to_change]
+            if self.soup_references[staff_name][field_to_change] is not None:
+                self.soup_references[staff_name][field_to_change].string = value
+            else:
+                self.soup_references[staff_name][field_to_change] = value
+
+                new_tag = self.soup.new_tag(field_to_change)
```

Comment:
I've removed trackName from two pieces in the unittest_metarepo and the insertion seems to be working.